pkgs/by-name/ca/caddy/package.nix

```diff
@@ -27,53 +30,124 @@ buildGoModule {
     hash = "sha256-CBfyqtWp3gYsYwaIxbfXO3AYaBiM7LutLC7uZgYXfkQ=";
   };
 
-  vendorHash = "sha256-1Api8bBZJ1/oYk4ZGIiwWCSraLzK9L+hsKXkFtk6iVM=";
-
   subPackages = [ "cmd/caddy" ];
 
   ldflags = [
-    "-s" "-w"
+    "-s"
+    "-w"
     "-X github.com/caddyserver/caddy/v2.CustomVersion=${version}"
   ];
 
   # matches upstream since v2.8.0
   tags = [ "nobadger" ];
 
-  nativeBuildInputs = [ installShellFiles ];
-
-  postInstall = ''
-    install -Dm644 ${dist}/init/caddy.service ${dist}/init/caddy-api.service -t $out/lib/systemd/system
-
-    substituteInPlace $out/lib/systemd/system/caddy.service \
-      --replace-fail "/usr/bin/caddy" "$out/bin/caddy"
-    substituteInPlace $out/lib/systemd/system/caddy-api.service \
-      --replace-fail "/usr/bin/caddy" "$out/bin/caddy"
-  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
-    # Generating man pages and completions fail on cross-compilation
-    # https://github.com/NixOS/nixpkgs/issues/308283
-
-    $out/bin/caddy manpage --directory manpages
-    installManPage manpages/*
-
-    installShellCompletion --cmd caddy \
-      --bash <($out/bin/caddy completion bash) \
-      --fish <($out/bin/caddy completion fish) \
-      --zsh <($out/bin/caddy completion zsh)
-  '';
-
-  passthru.tests = {
-    inherit (nixosTests) caddy;
-    version = testers.testVersion {
-      command = "${caddy}/bin/caddy version";
-      package = caddy;
-    };
-  };
+  nativeBuildInputs = [
+    gnused
```

Comment:
gnused shouldn't be necessary for non-extended caddy. Even in the `withPlugins` case, it doesn't seem strictly necessary (builds fine without it) though I don't know whether if it's best practice to include it anyway.

Reply:
`gnused` is necessary to support Mac build targets, due to Mac using BSD `sed` by default, which has a different syntax for `sed -i`. See [this review of the previous PR](https://github.com/NixOS/nixpkgs/pull/259275/files/a33b02fa9d664f31dadc8a874eb1a5dbaa9f4ecf#r1463001643) and [this thread](https://stackoverflow.com/questions/4247068/sed-command-with-i-option-failing-on-mac-but-works-on-linux).